lib/iris/tests/__init__.py

```diff
@@ -702,13 +700,17 @@ def check_graphic(self, tol=None):
                     if err.errno != 17:
                         raise
 
-            figure.savefig(result_fname)
+            def save_figure_hash():
+                figure = plt.gcf()
+                figure.savefig(result_fname)
```

Comment:
@marqh `plt.gcf().savefig(result_fname)` works for me ... but I don't see the point in making such a cosmetic change here in this PR.

I'm going to have to rebase #2192 with this anyways, so I can tweak away in my PR :wink: